cmd/fleetctl/apply.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/kolide/fleet/server/kolide"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+type specMetadata struct {
+	Kind    string      `json:"kind"`
+	Version string      `json:"apiVersion"`
+	Spec    interface{} `json:"spec"`
+}
+
+type specGroup struct {
+	Queries []*kolide.QuerySpec
+	Packs   []*kolide.PackSpec
+	Labels  []*kolide.LabelSpec
+	Options *kolide.OptionsSpec
+}
+
+func specGroupFromBytes(b []byte) (*specGroup, error) {
+	specs := &specGroup{
+		Queries: []*kolide.QuerySpec{},
+		Packs:   []*kolide.PackSpec{},
+		Labels:  []*kolide.LabelSpec{},
+	}
+
+	for _, spec := range strings.Split(string(b), "---") {
+		if strings.TrimSpace(spec) == "" {
+			continue
+		}
+
+		var s specMetadata
+		if err := yaml.Unmarshal([]byte(spec), &s); err != nil {
+			return nil, errors.Wrap(err, "error unmarshaling spec")
```

Comment:
I like to leave "error" out of the messages in `errors.Wrap` because we end up with `error foo....: error bar...: error baz....`. Usually we log these in a way that makes it clear an error occurred anyway.